docs/reference/objects/product/index.md

```diff
@@ -251,6 +258,7 @@ number
 {: .label .fs-1 }
 
 The sum of remaining stock for a product's variants, if any of the variants have infinite inventory this will return `nil`.
+If a product is a recurring experience, this returns the remaining stock across all dates.
```

Comment:
This one feels weird to me. I don't see when this would ever be useful. I wonder if that would be an expensive query to get the sum of the stock for every variant on every date.

Perhaps we update all the stock-related methods so they use the "default date" logic (next upcoming or most recent) and then deprecate them, encouraging theme developers to use stock methods on the ExperienceDateDrop instead:

- has_infinite_stock
- remaining_stock
- sold_out

Reply:
As discussed, stock across all dates is preferred, if unlikely to be used.